Ginger/GingerCoreCommon/Repository/RepositoryFolder.cs

```diff
@@ -620,7 +620,7 @@ public override RepositoryFolderBase AddSubFolder(string folderName)
             {
                 GetSubFolders();//calling it so it will find exisitng folders before adding new one
                 Directory.CreateDirectory(PathHelper.GetLongPath(FullPath));
-                GetSubFolders().Add(subfolder);
+                mSubFoldersCache?.Add(subfolder);
```

Comment:
we should not use "?" in code, it just not readable...
why we check if it is null? in which case it can be null? and if it is null so we won't add the folder?
also i was not able to reproduce the bug by following the ticket steps...

Reply:
? Is newly added by c# and it's simplifies the code to have null check. Instead of writing full null check with if and curly braces. I thing it is more readable we just need to get used to it.

In this case i agree null check is not needed. I just added it as additional check

Issue is not reproducible on existing folders. If you add new sub folder and try to move item to it it won't work. Because newly added folder is not found in cache. If reloading solution it will work